### PR TITLE
Update keystore

### PR DESCRIPTION
**Pull request by RnbWd**
* manually link react-native-keychain with warmup turned off
* remove `react-native-fs` (replaced with `rn-fetch-blob`)
* created keypair reducer with `{publicKey, secretKey}` (publicKey is signingKey)
* replaced the keychain methods for accessing secretKey with a keychain storage adapter for redux-persist
* `secretKey` is now accessible as a Uint8Array via redux (keypair reducer)
* removed publicKey from user reducer
* use `id` to determine app state is loaded instead of `publicKey`
* check if `id` is empty during bootstrap, delete all data if `id` is empty
* after second bootstrap, delete old redux-persist storage 
* create `photoDirectory` constant in @/utils/filesystem
* update redux-persist migration, have every sub-reducer register it's own storage configuration for persistence
* fix migration bug so that async migrate functions work - copy / pasted `createMigrate` from `redux-persist` but handle async functions properly
* use our own version of redux-persist-filesystem - simplified version of original library

### CODE REVIEW

File: BrightID/src/store/storage/index.js [hunk 1]
```diff
@@ -0,0 +1,5 @@
+// @flow
+
+export AsyncStorage from './asyncStorageAdapter';
+export FsStorage from './fsStorageAdapter';
+export KeychainStorage from './keychainAdapter';
```
Comment (TripleSpeeder) [inline BrightID/src/store/storage/index.js:5]: flow checker freaks out with this file:
```
Error:(3, 8) Unexpected identifier, expected the token `{`
Error:(3, 26) Missing comma between export specifiers
Error:(3, 26) Unexpected string, expected an identifier
Error:(3, 49) Missing comma between export specifiers
Error:(3, 49) Unexpected token `;`, expected an identifier
Error:(4, 1) Missing comma between export specifiers
Error:(4, 8) Missing comma between export specifiers
Error:(4, 18) Missing comma between export specifiers
Error:(4, 18) Duplicate export for `from`
Error:(4, 23) Missing comma between export specifiers
Error:(4, 23) Unexpected string, expected an identifier
Error:(4, 43) Missing comma between export specifiers
Error:(4, 43) Unexpected token `;`, expected an identifier
Error:(5, 1) Missing comma between export specifiers
Error:(5, 1) Duplicate export for `export`
Error:(5, 8) Missing comma between export specifiers
Error:(5, 24) Missing comma between export specifiers
Error:(5, 24) Duplicate export for `from`
Error:(5, 29) Missing comma between export specifiers
Error:(5, 29) Unexpected string, expected an identifier
Error:(5, 48) Missing comma between export specifiers
Error:(5, 48) Unexpected token `;`, expected an identifier
Error:(6, 1) Unexpected end of input, expected the token `}`
Error:(4, 1) Unexpected token `export`
Error:(5, 1) Unexpected token `export`
```
I think it requires named exports.

File: BrightID/flow-typed/myLibDef.js [hunk 2]
```diff
@@ -119,6 +120,11 @@ declare type GroupsState = {
   searchOpen: boolean,
 };
 
+declare type KeyPair = {
+  publicKey: string,
+  secretKey: Uint8Array,
+};
```
[the remark this answers is not in the document]
Comment (RnbWd) [reply BrightID/flow-typed/myLibDef.js:126]: but we would need to manually convert the secretKey everytime we use it. I don't know if we're using the publicKey anywhere, but if are using it, we should probably make both Uint8Array

File: BrightID/src/store/migrations/asyncCreateMigrate.js [hunk 3]
```diff
@@ -0,0 +1,55 @@
+// @flow
+
+import type { PersistedState, MigrationManifest } from 'redux-persist';
+import { VERSION } from '@/utils/constants';
```
Comment (TripleSpeeder) [inline BrightID/src/store/migrations/asyncCreateMigrate.js:4]: There is no `VERSION` exported from @utils/constants

**Review by TripleSpeeder — CHANGES_REQUESTED** [review]

File: BrightID/flow-typed/myLibDef.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (RnbWd) [reply BrightID/flow-typed/myLibDef.js:126]: we are currently only using the publicKey as a string, and i want to keep the reducer as close to what we're using in the app without additional steps of conversion

File: BrightID/src/store/storage/index.js (hunk 1 above)
Comment (RnbWd) [reply BrightID/src/store/storage/index.js:5]: i just removed flow

**Pull request merged**